providers/stores/abstractAzblobStore.js

```diff
@@ -64,13 +65,53 @@ class AbstractAzBlobStore {
     }
   }
 
+  /**
+   * Query and return the objects based on the query
+   *
+   * @param {object} query - The filters and sorts for the request
+   * @returns The data and continuationToken if there is more results
+   */
+  async find(query, continuationToken = '') {
+    const prefix = this._buildPrefix(query)
+    const result = await promisify(this.blobService.listBlobsSegmentedWithPrefix).bind(this.blobService)(
+      this.containerName,
+      prefix,
+      continuationToken,
+      {
+        include: azure.BlobUtilities.BlobListingDetails.METADATA
+      }
+    )
+    const data = []
+    result.entries
+      .forEach(entry => {
+        const path = entry.metadata.id
+        if (!path) return
+        data.push(this.get(EntityCoordinates.fromString(path)))
+      })
+      .filter(x => x)
+    return { data, continuationToken: result.continuationToken }
+  }
+
   _toStoragePathFromCoordinates(coordinates) {
     return AbstractFileStore.toStoragePathFromCoordinates(coordinates)
   }
 
   _toResultCoordinatesFromStoragePath(path) {
     return AbstractFileStore.toResultCoordinatesFromStoragePath(path)
   }
+
+  _buildPrefix(parameters) {
+    let result = ''
+    if (!parameters.type) return result
+    result += `${parameters.type}/`
```

Comment:
interesting problem. Since we can only do prefix in blob listing all the parts need to be provided.

Meta: I wonder if all providers need to implement `find`. Perhaps here we just throw saying "find is not supported"